docs/getting-started/concepts/dataset.md

```diff
@@ -0,0 +1,42 @@
+# Dataset
+
+Feast Datasets allow for conveniently saving dataframes that include both features and entities to be subsequently used for data analysis and model training.
```

Comment:
we should be consistent in casing with Dataset vs dataset here. probably just lower case throughout unless its code to be consistent?